src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,195 @@
+import React, {
+  useState,
+  useCallback,
+  useEffect,
+} from 'react';
+import {
+  getCommentsByPostId,
+  deleteComment,
+} from '../../api/comments';
+import { NewCommentForm } from '../NewCommentForm';
+import { Loader } from '../Loader';
+import { ErrorNotification } from '../ErrorNotification';
+import { Post } from '../../types/Post';
+import { Comment } from '../../types/Comment';
+import { Error } from '../../types/Error';
+
+type Props = {
+  posts: Post[];
+  postId: number | null;
+  error: Error | null;
+  onError: (error: Error | null) => void;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  posts,
+  postId,
+  error,
+  onError,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isCommentForm, setIsCommentForm] = useState(false);
+
+  const currentPost = posts.filter(post => post.id === postId)[0];
```

Comment:
Better to use [useMemo](https://www.w3schools.com/react/react_usememo.asp) to avoid additional recalculation